### ISSUE TITLE
New compiler: SCMD_DYNAMICBOUNDS not written when dynamic array is accessed with a literal index

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2081]
CC @fernewelten 

If you have a dynamic array in script, and access its element with a numeric literal, the compiler does not write SCMD_DYNAMICBOUNDS operation, which lets script to access any random position in engine's memory through this array, it seems.

Example:
```
function game_start() {
    int arr[] = new int[1];
    Display("%d", arr[1000]); // no error, prints random garbage

    int index = 1000;
    Display("%d", arr[index]); // fails with "out of bounds" error, as expected
}
```

Running engine under debugger shows that SCMD_DYNAMICBOUNDS is simply never executed on that line.

Debugging compiler shows that it's never written for expected statement, but I did not investigate further.

**Comment by fernewelten** [comment]
Looking into it.

**Comment by ivan-mogilko** [comment]
Fixed by #2092
